app/assets/javascripts/welcome.js.erb

```diff
@@ -1,121 +1,152 @@
-  var map, infoWindow;
-  var markers = [];
-  var savedmarkers= [];
-  var firstmarker= true;
-  var topic= '<%=image_url('0.png')%>';
-  function initMap() {
+var map, infoWindow;
+var markers = [];
+var markerstodelete = [];
```

Comment:
It´s better to keep a style when naming variables and functions. It´s recommended to use CamelCase , in this case: markersToDelete, same for firstMarker.